library/src/main/java/com/github/pwittchen/reactivenetwork/library/network/observing/strategy/PreLollipopNetworkObservingStrategy.java

```diff
@@ -75,17 +77,21 @@ private void tryToUnregisterReceiver(final Context context, final BroadcastRecei
     Log.e(LOG_TAG, message, exception);
   }
 
-  private Subscription unsubscribeInUiThread(final Action0 unsubscribe) {
-    return Subscriptions.create(new Action0() {
-      @Override public void call() {
+  private Disposable unsubscribeInUiThread(final Action unsubscribe) {
+    return Disposables.fromAction(new Action() {
+      @Override public void run() throws Exception {
         if (Looper.getMainLooper() == Looper.myLooper()) {
-          unsubscribe.call();
+          unsubscribe.run();
         } else {
           final Scheduler.Worker inner = AndroidSchedulers.mainThread().createWorker();
-          inner.schedule(new Action0() {
-            @Override public void call() {
-              unsubscribe.call();
-              inner.unsubscribe();
+          inner.schedule(new Runnable() {
+            @Override public void run() {
+              try {
+                unsubscribe.run();
+              } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
Please, do not print the whole StackTrace. Call `onError(message, exception)` method with an appropriate message.